pkg/component/component.go

```diff
@@ -119,23 +123,143 @@ func ApplyConfig(client kclient.ClientInterface, envSpecificInfo envinfo.EnvSpec
 	})
 }
 
+// ListAllNonDevfileComponents lists all NON Devfile Components from the given namespace
+// We then return a "dummy" simplified list of "components" only meant for basic listing output
+func ListAllNonDevfileComponents(client kclient.ClientInterface, namespace string) ([]Component, error) {
+	var deploymentList []v1.Deployment
+	var components []Component
+
+	deploymentList, err := client.ListDeploymentsByNamespaceAndSelector(namespace, project.GetNonOdoSelector())
+	if err != nil {
+		return []Component{}, errors.Wrapf(err, "unable to list components")
+	}
+
+	// Create a list of simplified components using Kubernetes data of non-odo related components
+	// We try to get all information from the "reccomended labels" document on the Kubernetes site:
+	// https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/
+	for _, elem := range deploymentList {
+
+		// First, let's get the name. For compatibility, some older tools do NOT use the "instance" label within Kubernetes,
+		// if name is empty, use the metadata name instead
```

Comment:
If something doesn't have `app.kubernetes.io/instance`  or `app.kubernetes.io/name` it should not be considered a component and it should not be displayed.